### ISSUE TITLE
Opening new notebook causes source edit box to not take input

### ISSUE THREAD

**Issue by mosiman** [mosiman/jupytui#1]

**Comment by mosiman** [comment]
There's something weird that I don't quite understand, but I have a workaround at least. 

The setup is that we have 

```
loop.widget === StatefulFrame( ... )
```

When we open a new notebook, a signal is sent to recreate a new `StatefulFrame` object, and assign the `loop.widget` object to this new `StatefulFrame`.

Before opening the new notebook, the focus widgets under the 'Nav' state would be something like 
```
[<ListBox selectable ... >, <Cell ... >, <AttrMap < LineBox <AttrMap < SelectableEdit ... > > > >]
```
This is more or less expected, although I would imagine that in determining focus hierarchies, `AttrMaps` should be skipped over. Upon entering the 'Insert' state, the focus widgets would be something like 
```
[<ListBox selectable ... >, <Cell selectable>, <AttrMap selectable< LineBox selectable < AttrMap selectable <SelectableEdit selectable ... > > > > ]
```
This is expected, because now the keypresses should end up going to the `SelectableEdit` widget (which should really be just an `Edit` but that's a different issue). 

Upon opening the new notebook, even on insert mode the `Cell` object is not selectable. Upon further inspection, the `SelectableEdit` widget IS selectable, but the `Cell` object isn't passing keystrokes through. 

This was a long debugging process. In the end, overriding the following methods
- `get_focus`
- `focus` (property decorator)
- `selectable`

fixed this weird issue. 

This is a bit of a duct tape fix, since the reason that everything works on the first load is unknown to me.

**Comment by mosiman** [comment]
fixed in e7bc641
